app/controllers/article_controller.rb

```diff
@@ -208,4 +214,13 @@ def eds_authenticated_user?
   def has_search_parameters?
     params[:q].present? || params[:f].present?
   end
+
+  def extract_fulltext_link(document, type)
+    links = document.fetch('eds_fulltext_links', [])
+    links.each do |link|
+      next if link['url'].blank? || link['url'] == 'detail'
+      return link['url'] if link['type'] == type.to_s
+    end
+    raise ArgumentError, "Missing #{type} fulltext link in document #{document.id}"
```

Comment:
Ok, below is the change for using a flash message. Any preference on the error message itself, right now it's:

> Full text link is unavailable at this time.

![fulltext-error mov](https://user-images.githubusercontent.com/1861171/29188791-2b7a40bc-7dc9-11e7-894e-e670e837112a.gif)